src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/Components/StreamFactory.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Server.Kestrel.Http;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Infrastructure
+{
+    public class StreamFactory : DefaultComponentFactory<Streams>
```

Comment:
Should the default factory throw if you try to set `MaxPooled` to anything other than zero?